src/components/userGroups/K8sObjectPermissions/K8sPermissionModal.tsx

```diff
@@ -0,0 +1,153 @@
+    const [isDataFilled, setIsDataFilled] = useState<boolean>(true)
+
+    useEffect(() => {
+        const disbale = k8PermissionList
+            ? k8PermissionList.find((item) => item.resource === null || item.resource?.length === 0)
+            : true
+        setIsDataFilled(!!disbale)
+    }, [k8PermissionList])
+
+    const handleK8sPermission = (action: string, key?: number, data?: any) => {
+        switch (action) {
+            case 'add':
+                k8PermissionList.splice(0, 0, getEmptyPermissionObject(k8PermissionList.length))
+                break
+            case 'delete':
+                k8PermissionList.splice(key, 1)
+                break
+            case 'clone':
+                k8PermissionList.splice(0, 0, { ...k8PermissionList[key], key: k8PermissionList.length })
+                break
+            case 'edit':
+                k8PermissionList[key].cluster = data
+                break
+            case 'onClusterChange':
+                k8PermissionList[key].cluster = data
+                k8PermissionList[key].namespace = null
+                k8PermissionList[key].group = null
+                k8PermissionList[key].kind = null
+                k8PermissionList[key].resource = null
+                break
+            case 'onNamespaceChange':
+                k8PermissionList[key].namespace = data
+                k8PermissionList[key].group = null
+                k8PermissionList[key].kind = null
+                k8PermissionList[key].resource = null
+                break
+            case 'onApiGroupChange':
+                k8PermissionList[key].group = data
+                k8PermissionList[key].kind = null
+                k8PermissionList[key].resource = null
+                break
+            case 'onKindChange':
+                k8PermissionList[key].kind = data
+                k8PermissionList[key].resource = null
+                break
+            case 'onObjectChange':
+                k8PermissionList[key].resource = data
+                break
+            case 'onRoleChange':
+                k8PermissionList[key].action = data
+                break
+            default:
+                break
+        }
+
+        setPermissionList([...k8PermissionList])
+    }
+
+    const stopPropogation = (e) => {
+        e.stopPropagation()
+    }
+
+    const addNewPermissionCard = () => {
+        handleK8sPermission('add')
+    }
+
+    const savePermission = () => {
+        setK8sPermission((prev) => {
+            if (selectedPermissionAction?.action === 'edit') {
+                prev[selectedPermissionAction.index] = k8PermissionList[0]
+                return [...prev]
+            }
+            return [...prev, ...k8PermissionList]
+        })
+        close(false)
+    }
+
+    return (
+        <VisibleModal className="" close={close}>
+            <div onClick={stopPropogation} className="modal-body--ci-material h-100 dc__overflow-hidden">
```

Comment:
Create a common class for the modal body instead of using the `modal-body--ci-material` one. Also, use `Drawer` instead of VisibleModal.

Reply:
done